DMX.Sdk.Tests.Unit/Services/Foundations/LabCommands/LabCommandServiceTests.Exceptions.Add.cs

```diff
@@ -0,0 +1,259 @@
+            string randomMessage = GetRandomString();
+            var httpMessage = new HttpResponseMessage();
+
+            var httpResponseException =
+                new HttpResponseException(httpMessage, randomMessage);
+
+            var failedLabCommandDependencyException
+                = new FailedLabCommandDependencyException(httpResponseException);
+
+            var expectedLabCommandDependencyException =
+                new LabCommandDependencyException(failedLabCommandDependencyException);
+
+            this.dmxApiBrokerMock.Setup(broker =>
+                broker.PostLabCommandAsync(It.IsAny<LabCommand>()))
+                    .ThrowsAsync(httpResponseException);
+
+            // when
+            ValueTask<LabCommand> addLabCommandTask =
+                this.labCommandService.AddLabCommandAsync(someLabCommand);
+
+            LabCommandDependencyException actualLabCommandDependencyException =
+                await Assert.ThrowsAsync<LabCommandDependencyException>(
+                    addLabCommandTask.AsTask);
+
+            // then
+            actualLabCommandDependencyException.Should()
+                .BeEquivalentTo(expectedLabCommandDependencyException);
+
+            this.dmxApiBrokerMock.Verify(broker =>
+                broker.PostLabCommandAsync(It.IsAny<LabCommand>()),
+                    Times.Once);
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogError(It.Is(SameExceptionAs(
+                    expectedLabCommandDependencyException))),
+                        Times.Once);
+
+            this.dmxApiBrokerMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+        }
+
+        [Fact]
+        public async Task ShouldThrowDependencyValidationExceptionOnAddIfBadRequestErrorOccursAndLogItAsync()
+        {
+            // given
+            LabCommand someLabCommand = CreateRandomLabCommand();
+            string randomMessage = GetRandomString();
+            var httpMessage = new HttpResponseMessage();
+
+            Dictionary<string, List<string>> randomDictionary =
+                CreateRandomDictionary();
+
+            var httpBadRequestException =
+                new HttpResponseBadRequestException(
+                    httpMessage,
+                    randomMessage);
+
+            httpBadRequestException.AddData(randomDictionary);
+
+            var invalidPostException =
+                new InvalidLabCommandException(
+                    httpBadRequestException,
+                    randomDictionary);
+
+            var expectedLabCommandDependencyValidationException =
+                new LabCommandDependencyValidationException(invalidPostException);
+
+            this.dmxApiBrokerMock.Setup(broker =>
+                broker.PostLabCommandAsync(It.IsAny<LabCommand>()))
+                    .ThrowsAsync(httpBadRequestException);
+
+            // when
+            ValueTask<LabCommand> postLabCommandTask =
```

Comment:
The variable name should match the method name. In this case here it should be `addLabCommandTask`